programs/system/CancelOrder.go

```diff
@@ -0,0 +1,227 @@
+	ag_format "github.com/bloXroute-Labs/solana-go/text/format"
+	ag_binary "github.com/gagliardetto/binary"
+	bin "github.com/gagliardetto/binary"
+	ag_treeout "github.com/gagliardetto/treeout"
+)
+
+// CancelOrder (V2)
+type CancelOrder struct {
+	Side    *serum.Side
+	OrderId *bin.Uint128
+
+	// 0. `[writable]` market
+	// 1. `[writable]` bids
+	// 2. `[writable]` asks
+	// 3. `[writable]` OpenOrders
+	// 4. `[signer]` the OpenOrders owner
+	// 5. `[writable]` event_q
+	ag_solanago.AccountMetaSlice `bin:"-" borsh_skip:"true"`
+}
+
+// NewCancelOrderInstructionBuilder creates a new `CancelOrder` instruction builder.
+func NewCancelOrderInstructionBuilder() *CancelOrder {
+	nd := &CancelOrder{
+		AccountMetaSlice: make(ag_solanago.AccountMetaSlice, 6),
+	}
+	return nd
+}
+
+func (c *CancelOrder) SetSide(s serum.Side) *CancelOrder {
+	c.Side = &s
+	return c
+}
+
+func (c *CancelOrder) SetOrderId(o bin.Uint128) *CancelOrder {
+	c.OrderId = &o
+	return c
+}
+
+func (c *CancelOrder) SetMarketAccount(a ag_solanago.PublicKey) *CancelOrder {
+	c.AccountMetaSlice[0] = ag_solanago.Meta(a).WRITE()
+	return c
+}
+
+func (c *CancelOrder) SetBidsAccount(a ag_solanago.PublicKey) *CancelOrder {
+	c.AccountMetaSlice[1] = ag_solanago.Meta(a).WRITE()
+	return c
+}
+
+func (c *CancelOrder) SetAsksAccount(a ag_solanago.PublicKey) *CancelOrder {
+	c.AccountMetaSlice[2] = ag_solanago.Meta(a).WRITE()
+	return c
+}
+
+func (c *CancelOrder) SetOpenOrdersAccount(a ag_solanago.PublicKey) *CancelOrder {
+	c.AccountMetaSlice[3] = ag_solanago.Meta(a).WRITE()
+	return c
+}
+
+func (c *CancelOrder) SetOwnerAccount(a ag_solanago.PublicKey) *CancelOrder {
+	c.AccountMetaSlice[4] = ag_solanago.Meta(a).WRITE().SIGNER()
+	return c
+}
+
+func (c *CancelOrder) SetEventQueueAccount(a ag_solanago.PublicKey) *CancelOrder {
+	c.AccountMetaSlice[5] = ag_solanago.Meta(a).WRITE()
+	return c
+}
+
+func (c *CancelOrder) MarketAccount() *ag_solanago.AccountMeta {
+	return c.AccountMetaSlice[0]
+}
+
+func (c *CancelOrder) BidsAccount() *ag_solanago.AccountMeta {
+	return c.AccountMetaSlice[1]
+}
+
+func (c *CancelOrder) AsksAccount() *ag_solanago.AccountMeta {
+	return c.AccountMetaSlice[2]
+}
+
+func (c *CancelOrder) OpenOrdersAccount() *ag_solanago.AccountMeta {
+	return c.AccountMetaSlice[3]
+}
+
+func (c *CancelOrder) OwnerAccount() *ag_solanago.AccountMeta {
+	return c.AccountMetaSlice[4]
+}
+
+func (c *CancelOrder) EventQueueAccount() *ag_solanago.AccountMeta {
+	return c.AccountMetaSlice[5]
+}
+
+func (c *CancelOrder) Accounts() []*ag_solanago.AccountMeta {
+	return c.AccountMetaSlice
+}
+
+func (c *CancelOrder) Data() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	if err := bin.NewBinEncoder(buf).Encode(c); err != nil {
+		return nil, fmt.Errorf("unable to encode CancelOrder (%v)", err)
+	}
+	return buf.Bytes(), nil
+}
+
+func (c *CancelOrder) ProgramID() solana.PublicKey {
+	// TODO: should there be a separate ProgramID for CancelOrder ?
+	return solana.SystemProgramID
```

Comment:
hm, pretty sure this should be the Serum Program ID?

Reply:
I haven't found a serum specific program ID, there's a file for these in `programs/serum/program.go` but it only has `DEXProgramIDV2`... would that be it ?